### ISSUE TITLE
"using assembly" can't find assembly in scriptdir on non-Windows because of bug in Parser.cs line:5136

### ISSUE THREAD

**Issue by JamesWTruher** [PowerShell/PowerShell#21161]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

create an assembly and run a script with the following content:
```powershell
using assembly mydll.dll
[mydll.mytype]
```

### Expected behavior

```console
no errors
```

### Actual behavior

```console
ParserError: /Users/james/bin/testit.ps1:1
Line |
   1 |  using assembly mydll.dll
     |                 ~~~~~~~~~~~~~~~~~~
     | Cannot load assembly 'mydll.dll'.
```

### Error details

line 5136: 

`assemblyFileName = workingDirectory + @"\" + assemblyFileName`

on a Unix System, `\` is not a directory separator but a valid filename character.

The fix is:

`assemblyFileName = Path.Combine(workingDirectory, assemblyFileName);`

### Environment data

```powershell
> $psversiontable

Name                           Value
----                           -----
PSVersion                      7.4.0
PSEdition                      Core
GitCommitId                    7.4.0
OS                             Darwin 23.3.0 Darwin Kernel Version 23.3.0: Wed Dec 20 21:28:58 PST 2023; root:xnu-10002.81.5~7/RELEASE_X86_64
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by JamesWTruher** [comment]
FWIW, there's a "simple" one line "fix" in this same location to actually address https://github.com/PowerShell/PowerShell/issues/18482 but I hesitate to suggest it (adding `Assembly.Load` a few lines lower right before the ast is returned). I'm not convinced that the parser should do such a thing as load an assembly during parsing and it seems ill advised for the parser to change the state of the process (even if it already does, why add _more_ bad actions). I haven't investigated further, but there _might_ be a way to do a metadata load but not sure what other follow-on impact that would have.

Also, we have some pending tests which I will likely re-enable (and add to) in the PR I'm preparing.

**Comment by daxian-dbw** [comment]
Resolved by #21169
